libs/net/services/Helpers/ApiService.cs

```diff
@@ -118,6 +118,41 @@ protected async Task<T> RetryRequestAsync<T>(Func<Task<T>> callbackDelegate)
             return await RetryRequestAsync<T>(callbackDelegate);
         }
     }
+
+    /// <summary>
+    /// Keep trying a request if the failure is caused by an optimistic concurrency error.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="callbackDelegate"></param>
+    /// <returns></returns>
+    public async Task<T> HandleConcurrencyAsync<T>(Func<Task<T>> callbackDelegate)
+    {
+        // Keep trying to update the record and handle concurrency errors.
+        while (true)
+        {
+            try
+            {
+                return await callbackDelegate();
+            }
+            catch (HttpClientRequestException ex)
+            {
+                // If it's a concurrency error, keep trying.  Otherwise throw the error.
+                this.Logger.LogError(ex, "Failed to complete request.  Determining if this is a concurrency error.");
+                var data = ex.Data["Body"] as string;
```

Comment:
I couldn't find a safe way to determine if the key exists  in the dictionary.  Regrettably .net uses a IDictionary object that doesn't have those features.